contracts/root/RootChain.sol

```diff
@@ -117,30 +108,28 @@ contract RootChain is Ownable, IRootChain {
     require(keccak256(dataList[1].toData()) == roundType, "Round type not same ");
     require(dataList[4].toByte() == voteType, "Vote type not same");
 
-    // check proposer
-    require(msg.sender == dataList[5].toAddress());
-
     // validate extra data using getSha256(extradata)
-    require(keccak256(dataList[6].toData()) == keccak256(bytes20(sha256(extradata))));
+    require(keccak256(dataList[5].toData()) == keccak256(bytes20(sha256(extradata))), "Extra data is invalid");
 
     // extract end and assign to current child
-    dataList = extradata.toRLPItem().toList()[0].toList();
-    uint256 start = currentChildBlock();
-    uint256 end = dataList[2].toUint();
-    bytes32 root = dataList[3].toBytes32();
+    dataList = extradata.toRLPItem().toList();
```

Comment:
does this work ? Dont we need to do like `extradata.toRLPItem().toList()[0].toList();` ?

Reply:
Yeah, it will work as I am only decoding msg data not basetx.